// @aws-sdk/client-textract/dist-cjs/protocols/Aws_json1_1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deserializeAws_json1_1StartExpenseAnalysisCommand = exports.deserializeAws_json1_1StartDocumentTextDetectionCommand = exports.deserializeAws_json1_1StartDocumentAnalysisCommand = exports.deserializeAws_json1_1GetExpenseAnalysisCommand = exports.deserializeAws_json1_1GetDocumentTextDetectionCommand = exports.deserializeAws_json1_1GetDocumentAnalysisCommand = exports.deserializeAws_json1_1DetectDocumentTextCommand = exports.deserializeAws_json1_1AnalyzeIDCommand = exports.deserializeAws_json1_1AnalyzeExpenseCommand = exports.deserializeAws_json1_1AnalyzeDocumentCommand = exports.serializeAws_json1_1StartExpenseAnalysisCommand = exports.serializeAws_json1_1StartDocumentTextDetectionCommand = exports.serializeAws_json1_1StartDocumentAnalysisCommand = exports.serializeAws_json1_1GetExpenseAnalysisCommand = exports.serializeAws_json1_1GetDocumentTextDetectionCommand = exports.serializeAws_json1_1GetDocumentAnalysisCommand = exports.serializeAws_json1_1DetectDocumentTextCommand = exports.serializeAws_json1_1AnalyzeIDCommand = exports.serializeAws_json1_1AnalyzeExpenseCommand = exports.serializeAws_json1_1AnalyzeDocumentCommand = void 0;
const protocol_http_1 = require("@aws-sdk/protocol-http");
const smithy_client_1 = require("@aws-sdk/smithy-client");
const models_0_1 = require("../models/models_0");
const TextractServiceException_1 = require("../models/TextractServiceException");
const serializeAws_json1_1AnalyzeDocumentCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.1",
        "x-amz-target": "Textract.AnalyzeDocument",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_1AnalyzeDocumentRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_1AnalyzeDocumentCommand = serializeAws_json1_1AnalyzeDocumentCommand;
const serializeAws_json1_1AnalyzeExpenseCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.1",
        "x-amz-target": "Textract.AnalyzeExpense",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_1AnalyzeExpenseRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_1AnalyzeExpenseCommand = serializeAws_json1_1AnalyzeExpenseCommand;
const serializeAws_json1_1AnalyzeIDCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.1",
        "x-amz-target": "Textract.AnalyzeID",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_1AnalyzeIDRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_1AnalyzeIDCommand = serializeAws_json1_1AnalyzeIDCommand;
const serializeAws_json1_1DetectDocumentTextCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.1",
        "x-amz-target": "Textract.DetectDocumentText",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_1DetectDocumentTextRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_1DetectDocumentTextCommand = serializeAws_json1_1DetectDocumentTextCommand;
const serializeAws_json1_1GetDocumentAnalysisCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.1",
        "x-amz-target": "Textract.GetDocumentAnalysis",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_1GetDocumentAnalysisRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_1GetDocumentAnalysisCommand = serializeAws_json1_1GetDocumentAnalysisCommand;
const serializeAws_json1_1GetDocumentTextDetectionCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.1",
        "x-amz-target": "Textract.GetDocumentTextDetection",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_1GetDocumentTextDetectionRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_1GetDocumentTextDetectionCommand = serializeAws_json1_1GetDocumentTextDetectionCommand;
const serializeAws_json1_1GetExpenseAnalysisCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.1",
        "x-amz-target": "Textract.GetExpenseAnalysis",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_1GetExpenseAnalysisRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_1GetExpenseAnalysisCommand = serializeAws_json1_1GetExpenseAnalysisCommand;
const serializeAws_json1_1StartDocumentAnalysisCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.1",
        "x-amz-target": "Textract.StartDocumentAnalysis",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_1StartDocumentAnalysisRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_1StartDocumentAnalysisCommand = serializeAws_json1_1StartDocumentAnalysisCommand;
const serializeAws_json1_1StartDocumentTextDetectionCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.1",
        "x-amz-target": "Textract.StartDocumentTextDetection",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_1StartDocumentTextDetectionRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_1StartDocumentTextDetectionCommand = serializeAws_json1_1StartDocumentTextDetectionCommand;
const serializeAws_json1_1StartExpenseAnalysisCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.1",
        "x-amz-target": "Textract.StartExpenseAnalysis",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_1StartExpenseAnalysisRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_1StartExpenseAnalysisCommand = serializeAws_json1_1StartExpenseAnalysisCommand;
const deserializeAws_json1_1AnalyzeDocumentCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_1AnalyzeDocumentCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = deserializeAws_json1_1AnalyzeDocumentResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_1AnalyzeDocumentCommand = deserializeAws_json1_1AnalyzeDocumentCommand;
const deserializeAws_json1_1AnalyzeDocumentCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "AccessDeniedException":
        case "com.amazonaws.textract#AccessDeniedException":
            throw await deserializeAws_json1_1AccessDeniedExceptionResponse(parsedOutput, context);
        case "BadDocumentException":
        case "com.amazonaws.textract#BadDocumentException":
            throw await deserializeAws_json1_1BadDocumentExceptionResponse(parsedOutput, context);
        case "DocumentTooLargeException":
        case "com.amazonaws.textract#DocumentTooLargeException":
            throw await deserializeAws_json1_1DocumentTooLargeExceptionResponse(parsedOutput, context);
        case "HumanLoopQuotaExceededException":
        case "com.amazonaws.textract#HumanLoopQuotaExceededException":
            throw await deserializeAws_json1_1HumanLoopQuotaExceededExceptionResponse(parsedOutput, context);
        case "InternalServerError":
        case "com.amazonaws.textract#InternalServerError":
            throw await deserializeAws_json1_1InternalServerErrorResponse(parsedOutput, context);
        case "InvalidParameterException":
        case "com.amazonaws.textract#InvalidParameterException":
            throw await deserializeAws_json1_1InvalidParameterExceptionResponse(parsedOutput, context);
        case "InvalidS3ObjectException":
        case "com.amazonaws.textract#InvalidS3ObjectException":
            throw await deserializeAws_json1_1InvalidS3ObjectExceptionResponse(parsedOutput, context);
        case "ProvisionedThroughputExceededException":
        case "com.amazonaws.textract#ProvisionedThroughputExceededException":
            throw await deserializeAws_json1_1ProvisionedThroughputExceededExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.textract#ThrottlingException":
            throw await deserializeAws_json1_1ThrottlingExceptionResponse(parsedOutput, context);
        case "UnsupportedDocumentException":
        case "com.amazonaws.textract#UnsupportedDocumentException":
            throw await deserializeAws_json1_1UnsupportedDocumentExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TextractServiceException_1.TextractServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_1AnalyzeExpenseCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_1AnalyzeExpenseCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = deserializeAws_json1_1AnalyzeExpenseResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_1AnalyzeExpenseCommand = deserializeAws_json1_1AnalyzeExpenseCommand;
const deserializeAws_json1_1AnalyzeExpenseCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "AccessDeniedException":
        case "com.amazonaws.textract#AccessDeniedException":
            throw await deserializeAws_json1_1AccessDeniedExceptionResponse(parsedOutput, context);
        case "BadDocumentException":
        case "com.amazonaws.textract#BadDocumentException":
            throw await deserializeAws_json1_1BadDocumentExceptionResponse(parsedOutput, context);
        case "DocumentTooLargeException":
        case "com.amazonaws.textract#DocumentTooLargeException":
            throw await deserializeAws_json1_1DocumentTooLargeExceptionResponse(parsedOutput, context);
        case "InternalServerError":
        case "com.amazonaws.textract#InternalServerError":
            throw await deserializeAws_json1_1InternalServerErrorResponse(parsedOutput, context);
        case "InvalidParameterException":
        case "com.amazonaws.textract#InvalidParameterException":
            throw await deserializeAws_json1_1InvalidParameterExceptionResponse(parsedOutput, context);
        case "InvalidS3ObjectException":
        case "com.amazonaws.textract#InvalidS3ObjectException":
            throw await deserializeAws_json1_1InvalidS3ObjectExceptionResponse(parsedOutput, context);
        case "ProvisionedThroughputExceededException":
        case "com.amazonaws.textract#ProvisionedThroughputExceededException":
            throw await deserializeAws_json1_1ProvisionedThroughputExceededExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.textract#ThrottlingException":
            throw await deserializeAws_json1_1ThrottlingExceptionResponse(parsedOutput, context);
        case "UnsupportedDocumentException":
        case "com.amazonaws.textract#UnsupportedDocumentException":
            throw await deserializeAws_json1_1UnsupportedDocumentExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TextractServiceException_1.TextractServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_1AnalyzeIDCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_1AnalyzeIDCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = deserializeAws_json1_1AnalyzeIDResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_1AnalyzeIDCommand = deserializeAws_json1_1AnalyzeIDCommand;
const deserializeAws_json1_1AnalyzeIDCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "AccessDeniedException":
        case "com.amazonaws.textract#AccessDeniedException":
            throw await deserializeAws_json1_1AccessDeniedExceptionResponse(parsedOutput, context);
        case "BadDocumentException":
        case "com.amazonaws.textract#BadDocumentException":
            throw await deserializeAws_json1_1BadDocumentExceptionResponse(parsedOutput, context);
        case "DocumentTooLargeException":
        case "com.amazonaws.textract#DocumentTooLargeException":
            throw await deserializeAws_json1_1DocumentTooLargeExceptionResponse(parsedOutput, context);
        case "InternalServerError":
        case "com.amazonaws.textract#InternalServerError":
            throw await deserializeAws_json1_1InternalServerErrorResponse(parsedOutput, context);
        case "InvalidParameterException":
        case "com.amazonaws.textract#InvalidParameterException":
            throw await deserializeAws_json1_1InvalidParameterExceptionResponse(parsedOutput, context);
        case "InvalidS3ObjectException":
        case "com.amazonaws.textract#InvalidS3ObjectException":
            throw await deserializeAws_json1_1InvalidS3ObjectExceptionResponse(parsedOutput, context);
        case "ProvisionedThroughputExceededException":
        case "com.amazonaws.textract#ProvisionedThroughputExceededException":
            throw await deserializeAws_json1_1ProvisionedThroughputExceededExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.textract#ThrottlingException":
            throw await deserializeAws_json1_1ThrottlingExceptionResponse(parsedOutput, context);
        case "UnsupportedDocumentException":
        case "com.amazonaws.textract#UnsupportedDocumentException":
            throw await deserializeAws_json1_1UnsupportedDocumentExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TextractServiceException_1.TextractServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_1DetectDocumentTextCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_1DetectDocumentTextCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = deserializeAws_json1_1DetectDocumentTextResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_1DetectDocumentTextCommand = deserializeAws_json1_1DetectDocumentTextCommand;
const deserializeAws_json1_1DetectDocumentTextCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "AccessDeniedException":
        case "com.amazonaws.textract#AccessDeniedException":
            throw await deserializeAws_json1_1AccessDeniedExceptionResponse(parsedOutput, context);
        case "BadDocumentException":
        case "com.amazonaws.textract#BadDocumentException":
            throw await deserializeAws_json1_1BadDocumentExceptionResponse(parsedOutput, context);
        case "DocumentTooLargeException":
        case "com.amazonaws.textract#DocumentTooLargeException":
            throw await deserializeAws_json1_1DocumentTooLargeExceptionResponse(parsedOutput, context);
        case "InternalServerError":
        case "com.amazonaws.textract#InternalServerError":
            throw await deserializeAws_json1_1InternalServerErrorResponse(parsedOutput, context);
        case "InvalidParameterException":
        case "com.amazonaws.textract#InvalidParameterException":
            throw await deserializeAws_json1_1InvalidParameterExceptionResponse(parsedOutput, context);
        case "InvalidS3ObjectException":
        case "com.amazonaws.textract#InvalidS3ObjectException":
            throw await deserializeAws_json1_1InvalidS3ObjectExceptionResponse(parsedOutput, context);
        case "ProvisionedThroughputExceededException":
        case "com.amazonaws.textract#ProvisionedThroughputExceededException":
            throw await deserializeAws_json1_1ProvisionedThroughputExceededExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.textract#ThrottlingException":
            throw await deserializeAws_json1_1ThrottlingExceptionResponse(parsedOutput, context);
        case "UnsupportedDocumentException":
        case "com.amazonaws.textract#UnsupportedDocumentException":
            throw await deserializeAws_json1_1UnsupportedDocumentExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TextractServiceException_1.TextractServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_1GetDocumentAnalysisCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_1GetDocumentAnalysisCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = deserializeAws_json1_1GetDocumentAnalysisResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_1GetDocumentAnalysisCommand = deserializeAws_json1_1GetDocumentAnalysisCommand;
const deserializeAws_json1_1GetDocumentAnalysisCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "AccessDeniedException":
        case "com.amazonaws.textract#AccessDeniedException":
            throw await deserializeAws_json1_1AccessDeniedExceptionResponse(parsedOutput, context);
        case "InternalServerError":
        case "com.amazonaws.textract#InternalServerError":
            throw await deserializeAws_json1_1InternalServerErrorResponse(parsedOutput, context);
        case "InvalidJobIdException":
        case "com.amazonaws.textract#InvalidJobIdException":
            throw await deserializeAws_json1_1InvalidJobIdExceptionResponse(parsedOutput, context);
        case "InvalidKMSKeyException":
        case "com.amazonaws.textract#InvalidKMSKeyException":
            throw await deserializeAws_json1_1InvalidKMSKeyExceptionResponse(parsedOutput, context);
        case "InvalidParameterException":
        case "com.amazonaws.textract#InvalidParameterException":
            throw await deserializeAws_json1_1InvalidParameterExceptionResponse(parsedOutput, context);
        case "InvalidS3ObjectException":
        case "com.amazonaws.textract#InvalidS3ObjectException":
            throw await deserializeAws_json1_1InvalidS3ObjectExceptionResponse(parsedOutput, context);
        case "ProvisionedThroughputExceededException":
        case "com.amazonaws.textract#ProvisionedThroughputExceededException":
            throw await deserializeAws_json1_1ProvisionedThroughputExceededExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.textract#ThrottlingException":
            throw await deserializeAws_json1_1ThrottlingExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TextractServiceException_1.TextractServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_1GetDocumentTextDetectionCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_1GetDocumentTextDetectionCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = deserializeAws_json1_1GetDocumentTextDetectionResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_1GetDocumentTextDetectionCommand = deserializeAws_json1_1GetDocumentTextDetectionCommand;
const deserializeAws_json1_1GetDocumentTextDetectionCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "AccessDeniedException":
        case "com.amazonaws.textract#AccessDeniedException":
            throw await deserializeAws_json1_1AccessDeniedExceptionResponse(parsedOutput, context);
        case "InternalServerError":
        case "com.amazonaws.textract#InternalServerError":
            throw await deserializeAws_json1_1InternalServerErrorResponse(parsedOutput, context);
        case "InvalidJobIdException":
        case "com.amazonaws.textract#InvalidJobIdException":
            throw await deserializeAws_json1_1InvalidJobIdExceptionResponse(parsedOutput, context);
        case "InvalidKMSKeyException":
        case "com.amazonaws.textract#InvalidKMSKeyException":
            throw await deserializeAws_json1_1InvalidKMSKeyExceptionResponse(parsedOutput, context);
        case "InvalidParameterException":
        case "com.amazonaws.textract#InvalidParameterException":
            throw await deserializeAws_json1_1InvalidParameterExceptionResponse(parsedOutput, context);
        case "InvalidS3ObjectException":
        case "com.amazonaws.textract#InvalidS3ObjectException":
            throw await deserializeAws_json1_1InvalidS3ObjectExceptionResponse(parsedOutput, context);
        case "ProvisionedThroughputExceededException":
        case "com.amazonaws.textract#ProvisionedThroughputExceededException":
            throw await deserializeAws_json1_1ProvisionedThroughputExceededExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.textract#ThrottlingException":
            throw await deserializeAws_json1_1ThrottlingExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TextractServiceException_1.TextractServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_1GetExpenseAnalysisCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_1GetExpenseAnalysisCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = deserializeAws_json1_1GetExpenseAnalysisResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_1GetExpenseAnalysisCommand = deserializeAws_json1_1GetExpenseAnalysisCommand;
const deserializeAws_json1_1GetExpenseAnalysisCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "AccessDeniedException":
        case "com.amazonaws.textract#AccessDeniedException":
            throw await deserializeAws_json1_1AccessDeniedExceptionResponse(parsedOutput, context);
        case "InternalServerError":
        case "com.amazonaws.textract#InternalServerError":
            throw await deserializeAws_json1_1InternalServerErrorResponse(parsedOutput, context);
        case "InvalidJobIdException":
        case "com.amazonaws.textract#InvalidJobIdException":
            throw await deserializeAws_json1_1InvalidJobIdExceptionResponse(parsedOutput, context);
        case "InvalidKMSKeyException":
        case "com.amazonaws.textract#InvalidKMSKeyException":
            throw await deserializeAws_json1_1InvalidKMSKeyExceptionResponse(parsedOutput, context);
        case "InvalidParameterException":
        case "com.amazonaws.textract#InvalidParameterException":
            throw await deserializeAws_json1_1InvalidParameterExceptionResponse(parsedOutput, context);
        case "InvalidS3ObjectException":
        case "com.amazonaws.textract#InvalidS3ObjectException":
            throw await deserializeAws_json1_1InvalidS3ObjectExceptionResponse(parsedOutput, context);
        case "ProvisionedThroughputExceededException":
        case "com.amazonaws.textract#ProvisionedThroughputExceededException":
            throw await deserializeAws_json1_1ProvisionedThroughputExceededExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.textract#ThrottlingException":
            throw await deserializeAws_json1_1ThrottlingExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TextractServiceException_1.TextractServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_1StartDocumentAnalysisCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_1StartDocumentAnalysisCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = deserializeAws_json1_1StartDocumentAnalysisResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_1StartDocumentAnalysisCommand = deserializeAws_json1_1StartDocumentAnalysisCommand;
const deserializeAws_json1_1StartDocumentAnalysisCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "AccessDeniedException":
        case "com.amazonaws.textract#AccessDeniedException":
            throw await deserializeAws_json1_1AccessDeniedExceptionResponse(parsedOutput, context);
        case "BadDocumentException":
        case "com.amazonaws.textract#BadDocumentException":
            throw await deserializeAws_json1_1BadDocumentExceptionResponse(parsedOutput, context);
        case "DocumentTooLargeException":
        case "com.amazonaws.textract#DocumentTooLargeException":
            throw await deserializeAws_json1_1DocumentTooLargeExceptionResponse(parsedOutput, context);
        case "IdempotentParameterMismatchException":
        case "com.amazonaws.textract#IdempotentParameterMismatchException":
            throw await deserializeAws_json1_1IdempotentParameterMismatchExceptionResponse(parsedOutput, context);
        case "InternalServerError":
        case "com.amazonaws.textract#InternalServerError":
            throw await deserializeAws_json1_1InternalServerErrorResponse(parsedOutput, context);
        case "InvalidKMSKeyException":
        case "com.amazonaws.textract#InvalidKMSKeyException":
            throw await deserializeAws_json1_1InvalidKMSKeyExceptionResponse(parsedOutput, context);
        case "InvalidParameterException":
        case "com.amazonaws.textract#InvalidParameterException":
            throw await deserializeAws_json1_1InvalidParameterExceptionResponse(parsedOutput, context);
        case "InvalidS3ObjectException":
        case "com.amazonaws.textract#InvalidS3ObjectException":
            throw await deserializeAws_json1_1InvalidS3ObjectExceptionResponse(parsedOutput, context);
        case "LimitExceededException":
        case "com.amazonaws.textract#LimitExceededException":
            throw await deserializeAws_json1_1LimitExceededExceptionResponse(parsedOutput, context);
        case "ProvisionedThroughputExceededException":
        case "com.amazonaws.textract#ProvisionedThroughputExceededException":
            throw await deserializeAws_json1_1ProvisionedThroughputExceededExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.textract#ThrottlingException":
            throw await deserializeAws_json1_1ThrottlingExceptionResponse(parsedOutput, context);
        case "UnsupportedDocumentException":
        case "com.amazonaws.textract#UnsupportedDocumentException":
            throw await deserializeAws_json1_1UnsupportedDocumentExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TextractServiceException_1.TextractServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_1StartDocumentTextDetectionCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_1StartDocumentTextDetectionCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = deserializeAws_json1_1StartDocumentTextDetectionResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_1StartDocumentTextDetectionCommand = deserializeAws_json1_1StartDocumentTextDetectionCommand;
const deserializeAws_json1_1StartDocumentTextDetectionCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "AccessDeniedException":
        case "com.amazonaws.textract#AccessDeniedException":
            throw await deserializeAws_json1_1AccessDeniedExceptionResponse(parsedOutput, context);
        case "BadDocumentException":
        case "com.amazonaws.textract#BadDocumentException":
            throw await deserializeAws_json1_1BadDocumentExceptionResponse(parsedOutput, context);
        case "DocumentTooLargeException":
        case "com.amazonaws.textract#DocumentTooLargeException":
            throw await deserializeAws_json1_1DocumentTooLargeExceptionResponse(parsedOutput, context);
        case "IdempotentParameterMismatchException":
        case "com.amazonaws.textract#IdempotentParameterMismatchException":
            throw await deserializeAws_json1_1IdempotentParameterMismatchExceptionResponse(parsedOutput, context);
        case "InternalServerError":
        case "com.amazonaws.textract#InternalServerError":
            throw await deserializeAws_json1_1InternalServerErrorResponse(parsedOutput, context);
        case "InvalidKMSKeyException":
        case "com.amazonaws.textract#InvalidKMSKeyException":
            throw await deserializeAws_json1_1InvalidKMSKeyExceptionResponse(parsedOutput, context);
        case "InvalidParameterException":
        case "com.amazonaws.textract#InvalidParameterException":
            throw await deserializeAws_json1_1InvalidParameterExceptionResponse(parsedOutput, context);
        case "InvalidS3ObjectException":
        case "com.amazonaws.textract#InvalidS3ObjectException":
            throw await deserializeAws_json1_1InvalidS3ObjectExceptionResponse(parsedOutput, context);
        case "LimitExceededException":
        case "com.amazonaws.textract#LimitExceededException":
            throw await deserializeAws_json1_1LimitExceededExceptionResponse(parsedOutput, context);
        case "ProvisionedThroughputExceededException":
        case "com.amazonaws.textract#ProvisionedThroughputExceededException":
            throw await deserializeAws_json1_1ProvisionedThroughputExceededExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.textract#ThrottlingException":
            throw await deserializeAws_json1_1ThrottlingExceptionResponse(parsedOutput, context);
        case "UnsupportedDocumentException":
        case "com.amazonaws.textract#UnsupportedDocumentException":
            throw await deserializeAws_json1_1UnsupportedDocumentExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TextractServiceException_1.TextractServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_1StartExpenseAnalysisCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_1StartExpenseAnalysisCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = deserializeAws_json1_1StartExpenseAnalysisResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_1StartExpenseAnalysisCommand = deserializeAws_json1_1StartExpenseAnalysisCommand;
const deserializeAws_json1_1StartExpenseAnalysisCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "AccessDeniedException":
        case "com.amazonaws.textract#AccessDeniedException":
            throw await deserializeAws_json1_1AccessDeniedExceptionResponse(parsedOutput, context);
        case "BadDocumentException":
        case "com.amazonaws.textract#BadDocumentException":
            throw await deserializeAws_json1_1BadDocumentExceptionResponse(parsedOutput, context);
        case "DocumentTooLargeException":
        case "com.amazonaws.textract#DocumentTooLargeException":
            throw await deserializeAws_json1_1DocumentTooLargeExceptionResponse(parsedOutput, context);
        case "IdempotentParameterMismatchException":
        case "com.amazonaws.textract#IdempotentParameterMismatchException":
            throw await deserializeAws_json1_1IdempotentParameterMismatchExceptionResponse(parsedOutput, context);
        case "InternalServerError":
        case "com.amazonaws.textract#InternalServerError":
            throw await deserializeAws_json1_1InternalServerErrorResponse(parsedOutput, context);
        case "InvalidKMSKeyException":
        case "com.amazonaws.textract#InvalidKMSKeyException":
            throw await deserializeAws_json1_1InvalidKMSKeyExceptionResponse(parsedOutput, context);
        case "InvalidParameterException":
        case "com.amazonaws.textract#InvalidParameterException":
            throw await deserializeAws_json1_1InvalidParameterExceptionResponse(parsedOutput, context);
        case "InvalidS3ObjectException":
        case "com.amazonaws.textract#InvalidS3ObjectException":
            throw await deserializeAws_json1_1InvalidS3ObjectExceptionResponse(parsedOutput, context);
        case "LimitExceededException":
        case "com.amazonaws.textract#LimitExceededException":
            throw await deserializeAws_json1_1LimitExceededExceptionResponse(parsedOutput, context);
        case "ProvisionedThroughputExceededException":
        case "com.amazonaws.textract#ProvisionedThroughputExceededException":
            throw await deserializeAws_json1_1ProvisionedThroughputExceededExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.textract#ThrottlingException":
            throw await deserializeAws_json1_1ThrottlingExceptionResponse(parsedOutput, context);
        case "UnsupportedDocumentException":
        case "com.amazonaws.textract#UnsupportedDocumentException":
            throw await deserializeAws_json1_1UnsupportedDocumentExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TextractServiceException_1.TextractServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_1AccessDeniedExceptionResponse = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = deserializeAws_json1_1AccessDeniedException(body, context);
    const exception = new models_0_1.AccessDeniedException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const deserializeAws_json1_1BadDocumentExceptionResponse = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = deserializeAws_json1_1BadDocumentException(body, context);
    const exception = new models_0_1.BadDocumentException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const deserializeAws_json1_1DocumentTooLargeExceptionResponse = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = deserializeAws_json1_1DocumentTooLargeException(body, context);
    const exception = new models_0_1.DocumentTooLargeException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const deserializeAws_json1_1HumanLoopQuotaExceededExceptionResponse = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = deserializeAws_json1_1HumanLoopQuotaExceededException(body, context);
    const exception = new models_0_1.HumanLoopQuotaExceededException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const deserializeAws_json1_1IdempotentParameterMismatchExceptionResponse = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = deserializeAws_json1_1IdempotentParameterMismatchException(body, context);
    const exception = new models_0_1.IdempotentParameterMismatchException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const deserializeAws_json1_1InternalServerErrorResponse = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = deserializeAws_json1_1InternalServerError(body, context);
    const exception = new models_0_1.InternalServerError({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const deserializeAws_json1_1InvalidJobIdExceptionResponse = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = deserializeAws_json1_1InvalidJobIdException(body, context);
    const exception = new models_0_1.InvalidJobIdException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const deserializeAws_json1_1InvalidKMSKeyExceptionResponse = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = deserializeAws_json1_1InvalidKMSKeyException(body, context);
    const exception = new models_0_1.InvalidKMSKeyException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const deserializeAws_json1_1InvalidParameterExceptionResponse = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = deserializeAws_json1_1InvalidParameterException(body, context);
    const exception = new models_0_1.InvalidParameterException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const deserializeAws_json1_1InvalidS3ObjectExceptionResponse = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = deserializeAws_json1_1InvalidS3ObjectException(body, context);
    const exception = new models_0_1.InvalidS3ObjectException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const deserializeAws_json1_1LimitExceededExceptionResponse = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = deserializeAws_json1_1LimitExceededException(body, context);
    const exception = new models_0_1.LimitExceededException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const deserializeAws_json1_1ProvisionedThroughputExceededExceptionResponse = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = deserializeAws_json1_1ProvisionedThroughputExceededException(body, context);
    const exception = new models_0_1.ProvisionedThroughputExceededException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const deserializeAws_json1_1ThrottlingExceptionResponse = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = deserializeAws_json1_1ThrottlingException(body, context);
    const exception = new models_0_1.ThrottlingException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const deserializeAws_json1_1UnsupportedDocumentExceptionResponse = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = deserializeAws_json1_1UnsupportedDocumentException(body, context);
    const exception = new models_0_1.UnsupportedDocumentException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const serializeAws_json1_1AnalyzeDocumentRequest = (input, context) => {
    return {
        ...(input.Document != null && { Document: serializeAws_json1_1Document(input.Document, context) }),
        ...(input.FeatureTypes != null && { FeatureTypes: serializeAws_json1_1FeatureTypes(input.FeatureTypes, context) }),
        ...(input.HumanLoopConfig != null && {
            HumanLoopConfig: serializeAws_json1_1HumanLoopConfig(input.HumanLoopConfig, context),
        }),
        ...(input.QueriesConfig != null && {
            QueriesConfig: serializeAws_json1_1QueriesConfig(input.QueriesConfig, context),
        }),
    };
};
const serializeAws_json1_1AnalyzeExpenseRequest = (input, context) => {
    return {
        ...(input.Document != null && { Document: serializeAws_json1_1Document(input.Document, context) }),
    };
};
const serializeAws_json1_1AnalyzeIDRequest = (input, context) => {
    return {
        ...(input.DocumentPages != null && {
            DocumentPages: serializeAws_json1_1DocumentPages(input.DocumentPages, context),
        }),
    };
};
const serializeAws_json1_1ContentClassifiers = (input, context) => {
    return input
        .filter((e) => e != null)
        .map((entry) => {
        return entry;
    });
};
const serializeAws_json1_1DetectDocumentTextRequest = (input, context) => {
    return {
        ...(input.Document != null && { Document: serializeAws_json1_1Document(input.Document, context) }),
    };
};
const serializeAws_json1_1Document = (input, context) => {
    return {
        ...(input.Bytes != null && { Bytes: context.base64Encoder(input.Bytes) }),
        ...(input.S3Object != null && { S3Object: serializeAws_json1_1S3Object(input.S3Object, context) }),
    };
};
const serializeAws_json1_1DocumentLocation = (input, context) => {
    return {
        ...(input.S3Object != null && { S3Object: serializeAws_json1_1S3Object(input.S3Object, context) }),
    };
};
const serializeAws_json1_1DocumentPages = (input, context) => {
    return input
        .filter((e) => e != null)
        .map((entry) => {
        return serializeAws_json1_1Document(entry, context);
    });
};
const serializeAws_json1_1FeatureTypes = (input, context) => {
    return input
        .filter((e) => e != null)
        .map((entry) => {
        return entry;
    });
};
const serializeAws_json1_1GetDocumentAnalysisRequest = (input, context) => {
    return {
        ...(input.JobId != null && { JobId: input.JobId }),
        ...(input.MaxResults != null && { MaxResults: input.MaxResults }),
        ...(input.NextToken != null && { NextToken: input.NextToken }),
    };
};
const serializeAws_json1_1GetDocumentTextDetectionRequest = (input, context) => {
    return {
        ...(input.JobId != null && { JobId: input.JobId }),
        ...(input.MaxResults != null && { MaxResults: input.MaxResults }),
        ...(input.NextToken != null && { NextToken: input.NextToken }),
    };
};
const serializeAws_json1_1GetExpenseAnalysisRequest = (input, context) => {
    return {
        ...(input.JobId != null && { JobId: input.JobId }),
        ...(input.MaxResults != null && { MaxResults: input.MaxResults }),
        ...(input.NextToken != null && { NextToken: input.NextToken }),
    };
};
const serializeAws_json1_1HumanLoopConfig = (input, context) => {
    return {
        ...(input.DataAttributes != null && {
            DataAttributes: serializeAws_json1_1HumanLoopDataAttributes(input.DataAttributes, context),
        }),
        ...(input.FlowDefinitionArn != null && { FlowDefinitionArn: input.FlowDefinitionArn }),
        ...(input.HumanLoopName != null && { HumanLoopName: input.HumanLoopName }),
    };
};
const serializeAws_json1_1HumanLoopDataAttributes = (input, context) => {
    return {
        ...(input.ContentClassifiers != null && {
            ContentClassifiers: serializeAws_json1_1ContentClassifiers(input.ContentClassifiers, context),
        }),
    };
};
const serializeAws_json1_1NotificationChannel = (input, context) => {
    return {
        ...(input.RoleArn != null && { RoleArn: input.RoleArn }),
        ...(input.SNSTopicArn != null && { SNSTopicArn: input.SNSTopicArn }),
    };
};
const serializeAws_json1_1OutputConfig = (input, context) => {
    return {
        ...(input.S3Bucket != null && { S3Bucket: input.S3Bucket }),
        ...(input.S3Prefix != null && { S3Prefix: input.S3Prefix }),
    };
};
const serializeAws_json1_1Queries = (input, context) => {
    return input
        .filter((e) => e != null)
        .map((entry) => {
        return serializeAws_json1_1Query(entry, context);
    });
};
const serializeAws_json1_1QueriesConfig = (input, context) => {
    return {
        ...(input.Queries != null && { Queries: serializeAws_json1_1Queries(input.Queries, context) }),
    };
};
const serializeAws_json1_1Query = (input, context) => {
    return {
        ...(input.Alias != null && { Alias: input.Alias }),
        ...(input.Pages != null && { Pages: serializeAws_json1_1QueryPages(input.Pages, context) }),
        ...(input.Text != null && { Text: input.Text }),
    };
};
const serializeAws_json1_1QueryPages = (input, context) => {
    return input
        .filter((e) => e != null)
        .map((entry) => {
        return entry;
    });
};
const serializeAws_json1_1S3Object = (input, context) => {
    return {
        ...(input.Bucket != null && { Bucket: input.Bucket }),
        ...(input.Name != null && { Name: input.Name }),
        ...(input.Version != null && { Version: input.Version }),
    };
};
const serializeAws_json1_1StartDocumentAnalysisRequest = (input, context) => {
    return {
        ...(input.ClientRequestToken != null && { ClientRequestToken: input.ClientRequestToken }),
        ...(input.DocumentLocation != null && {
            DocumentLocation: serializeAws_json1_1DocumentLocation(input.DocumentLocation, context),
        }),
        ...(input.FeatureTypes != null && { FeatureTypes: serializeAws_json1_1FeatureTypes(input.FeatureTypes, context) }),
        ...(input.JobTag != null && { JobTag: input.JobTag }),
        ...(input.KMSKeyId != null && { KMSKeyId: input.KMSKeyId }),
        ...(input.NotificationChannel != null && {
            NotificationChannel: serializeAws_json1_1NotificationChannel(input.NotificationChannel, context),
        }),
        ...(input.OutputConfig != null && { OutputConfig: serializeAws_json1_1OutputConfig(input.OutputConfig, context) }),
        ...(input.QueriesConfig != null && {
            QueriesConfig: serializeAws_json1_1QueriesConfig(input.QueriesConfig, context),
        }),
    };
};
const serializeAws_json1_1StartDocumentTextDetectionRequest = (input, context) => {
    return {
        ...(input.ClientRequestToken != null && { ClientRequestToken: input.ClientRequestToken }),
        ...(input.DocumentLocation != null && {
            DocumentLocation: serializeAws_json1_1DocumentLocation(input.DocumentLocation, context),
        }),
        ...(input.JobTag != null && { JobTag: input.JobTag }),
        ...(input.KMSKeyId != null && { KMSKeyId: input.KMSKeyId }),
        ...(input.NotificationChannel != null && {
            NotificationChannel: serializeAws_json1_1NotificationChannel(input.NotificationChannel, context),
        }),
        ...(input.OutputConfig != null && { OutputConfig: serializeAws_json1_1OutputConfig(input.OutputConfig, context) }),
    };
};
const serializeAws_json1_1StartExpenseAnalysisRequest = (input, context) => {
    return {
        ...(input.ClientRequestToken != null && { ClientRequestToken: input.ClientRequestToken }),
        ...(input.DocumentLocation != null && {
            DocumentLocation: serializeAws_json1_1DocumentLocation(input.DocumentLocation, context),
        }),
        ...(input.JobTag != null && { JobTag: input.JobTag }),
        ...(input.KMSKeyId != null && { KMSKeyId: input.KMSKeyId }),
        ...(input.NotificationChannel != null && {
            NotificationChannel: serializeAws_json1_1NotificationChannel(input.NotificationChannel, context),
        }),
        ...(input.OutputConfig != null && { OutputConfig: serializeAws_json1_1OutputConfig(input.OutputConfig, context) }),
    };
};
const deserializeAws_json1_1AccessDeniedException = (output, context) => {
    return {
        Code: (0, smithy_client_1.expectString)(output.Code),
        Message: (0, smithy_client_1.expectString)(output.Message),
    };
};
const deserializeAws_json1_1AnalyzeDocumentResponse = (output, context) => {
    return {
        AnalyzeDocumentModelVersion: (0, smithy_client_1.expectString)(output.AnalyzeDocumentModelVersion),
        Blocks: output.Blocks != null ? deserializeAws_json1_1BlockList(output.Blocks, context) : undefined,
        DocumentMetadata: output.DocumentMetadata != null
            ? deserializeAws_json1_1DocumentMetadata(output.DocumentMetadata, context)
            : undefined,
        HumanLoopActivationOutput: output.HumanLoopActivationOutput != null
            ? deserializeAws_json1_1HumanLoopActivationOutput(output.HumanLoopActivationOutput, context)
            : undefined,
    };
};
const deserializeAws_json1_1AnalyzeExpenseResponse = (output, context) => {
    return {
        DocumentMetadata: output.DocumentMetadata != null
            ? deserializeAws_json1_1DocumentMetadata(output.DocumentMetadata, context)
            : undefined,
        ExpenseDocuments: output.ExpenseDocuments != null
            ? deserializeAws_json1_1ExpenseDocumentList(output.ExpenseDocuments, context)
            : undefined,
    };
};
const deserializeAws_json1_1AnalyzeIDDetections = (output, context) => {
    return {
        Confidence: (0, smithy_client_1.limitedParseFloat32)(output.Confidence),
        NormalizedValue: output.NormalizedValue != null
            ? deserializeAws_json1_1NormalizedValue(output.NormalizedValue, context)
            : undefined,
        Text: (0, smithy_client_1.expectString)(output.Text),
    };
};
const deserializeAws_json1_1AnalyzeIDResponse = (output, context) => {
    return {
        AnalyzeIDModelVersion: (0, smithy_client_1.expectString)(output.AnalyzeIDModelVersion),
        DocumentMetadata: output.DocumentMetadata != null
            ? deserializeAws_json1_1DocumentMetadata(output.DocumentMetadata, context)
            : undefined,
        IdentityDocuments: output.IdentityDocuments != null
            ? deserializeAws_json1_1IdentityDocumentList(output.IdentityDocuments, context)
            : undefined,
    };
};
const deserializeAws_json1_1BadDocumentException = (output, context) => {
    return {
        Code: (0, smithy_client_1.expectString)(output.Code),
        Message: (0, smithy_client_1.expectString)(output.Message),
    };
};
const deserializeAws_json1_1Block = (output, context) => {
    return {
        BlockType: (0, smithy_client_1.expectString)(output.BlockType),
        ColumnIndex: (0, smithy_client_1.expectInt32)(output.ColumnIndex),
        ColumnSpan: (0, smithy_client_1.expectInt32)(output.ColumnSpan),
        Confidence: (0, smithy_client_1.limitedParseFloat32)(output.Confidence),
        EntityTypes: output.EntityTypes != null ? deserializeAws_json1_1EntityTypes(output.EntityTypes, context) : undefined,
        Geometry: output.Geometry != null ? deserializeAws_json1_1Geometry(output.Geometry, context) : undefined,
        Id: (0, smithy_client_1.expectString)(output.Id),
        Page: (0, smithy_client_1.expectInt32)(output.Page),
        Query: output.Query != null ? deserializeAws_json1_1Query(output.Query, context) : undefined,
        Relationships: output.Relationships != null ? deserializeAws_json1_1RelationshipList(output.Relationships, context) : undefined,
        RowIndex: (0, smithy_client_1.expectInt32)(output.RowIndex),
        RowSpan: (0, smithy_client_1.expectInt32)(output.RowSpan),
        SelectionStatus: (0, smithy_client_1.expectString)(output.SelectionStatus),
        Text: (0, smithy_client_1.expectString)(output.Text),
        TextType: (0, smithy_client_1.expectString)(output.TextType),
    };
};
const deserializeAws_json1_1BlockList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        if (entry === null) {
            return null;
        }
        return deserializeAws_json1_1Block(entry, context);
    });
    return retVal;
};
const deserializeAws_json1_1BoundingBox = (output, context) => {
    return {
        Height: (0, smithy_client_1.limitedParseFloat32)(output.Height),
        Left: (0, smithy_client_1.limitedParseFloat32)(output.Left),
        Top: (0, smithy_client_1.limitedParseFloat32)(output.Top),
        Width: (0, smithy_client_1.limitedParseFloat32)(output.Width),
    };
};
const deserializeAws_json1_1DetectDocumentTextResponse = (output, context) => {
    return {
        Blocks: output.Blocks != null ? deserializeAws_json1_1BlockList(output.Blocks, context) : undefined,
        DetectDocumentTextModelVersion: (0, smithy_client_1.expectString)(output.DetectDocumentTextModelVersion),
        DocumentMetadata: output.DocumentMetadata != null
            ? deserializeAws_json1_1DocumentMetadata(output.DocumentMetadata, context)
            : undefined,
    };
};
const deserializeAws_json1_1DocumentMetadata = (output, context) => {
    return {
        Pages: (0, smithy_client_1.expectInt32)(output.Pages),
    };
};
const deserializeAws_json1_1DocumentTooLargeException = (output, context) => {
    return {
        Code: (0, smithy_client_1.expectString)(output.Code),
        Message: (0, smithy_client_1.expectString)(output.Message),
    };
};
const deserializeAws_json1_1EntityTypes = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        if (entry === null) {
            return null;
        }
        return (0, smithy_client_1.expectString)(entry);
    });
    return retVal;
};
const deserializeAws_json1_1ExpenseDetection = (output, context) => {
    return {
        Confidence: (0, smithy_client_1.limitedParseFloat32)(output.Confidence),
        Geometry: output.Geometry != null ? deserializeAws_json1_1Geometry(output.Geometry, context) : undefined,
        Text: (0, smithy_client_1.expectString)(output.Text),
    };
};
const deserializeAws_json1_1ExpenseDocument = (output, context) => {
    return {
        ExpenseIndex: (0, smithy_client_1.expectInt32)(output.ExpenseIndex),
        LineItemGroups: output.LineItemGroups != null
            ? deserializeAws_json1_1LineItemGroupList(output.LineItemGroups, context)
            : undefined,
        SummaryFields: output.SummaryFields != null ? deserializeAws_json1_1ExpenseFieldList(output.SummaryFields, context) : undefined,
    };
};
const deserializeAws_json1_1ExpenseDocumentList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        if (entry === null) {
            return null;
        }
        return deserializeAws_json1_1ExpenseDocument(entry, context);
    });
    return retVal;
};
const deserializeAws_json1_1ExpenseField = (output, context) => {
    return {
        LabelDetection: output.LabelDetection != null
            ? deserializeAws_json1_1ExpenseDetection(output.LabelDetection, context)
            : undefined,
        PageNumber: (0, smithy_client_1.expectInt32)(output.PageNumber),
        Type: output.Type != null ? deserializeAws_json1_1ExpenseType(output.Type, context) : undefined,
        ValueDetection: output.ValueDetection != null
            ? deserializeAws_json1_1ExpenseDetection(output.ValueDetection, context)
            : undefined,
    };
};
const deserializeAws_json1_1ExpenseFieldList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        if (entry === null) {
            return null;
        }
        return deserializeAws_json1_1ExpenseField(entry, context);
    });
    return retVal;
};
const deserializeAws_json1_1ExpenseType = (output, context) => {
    return {
        Confidence: (0, smithy_client_1.limitedParseFloat32)(output.Confidence),
        Text: (0, smithy_client_1.expectString)(output.Text),
    };
};
const deserializeAws_json1_1Geometry = (output, context) => {
    return {
        BoundingBox: output.BoundingBox != null ? deserializeAws_json1_1BoundingBox(output.BoundingBox, context) : undefined,
        Polygon: output.Polygon != null ? deserializeAws_json1_1Polygon(output.Polygon, context) : undefined,
    };
};
const deserializeAws_json1_1GetDocumentAnalysisResponse = (output, context) => {
    return {
        AnalyzeDocumentModelVersion: (0, smithy_client_1.expectString)(output.AnalyzeDocumentModelVersion),
        Blocks: output.Blocks != null ? deserializeAws_json1_1BlockList(output.Blocks, context) : undefined,
        DocumentMetadata: output.DocumentMetadata != null
            ? deserializeAws_json1_1DocumentMetadata(output.DocumentMetadata, context)
            : undefined,
        JobStatus: (0, smithy_client_1.expectString)(output.JobStatus),
        NextToken: (0, smithy_client_1.expectString)(output.NextToken),
        StatusMessage: (0, smithy_client_1.expectString)(output.StatusMessage),
        Warnings: output.Warnings != null ? deserializeAws_json1_1Warnings(output.Warnings, context) : undefined,
    };
};
const deserializeAws_json1_1GetDocumentTextDetectionResponse = (output, context) => {
    return {
        Blocks: output.Blocks != null ? deserializeAws_json1_1BlockList(output.Blocks, context) : undefined,
        DetectDocumentTextModelVersion: (0, smithy_client_1.expectString)(output.DetectDocumentTextModelVersion),
        DocumentMetadata: output.DocumentMetadata != null
            ? deserializeAws_json1_1DocumentMetadata(output.DocumentMetadata, context)
            : undefined,
        JobStatus: (0, smithy_client_1.expectString)(output.JobStatus),
        NextToken: (0, smithy_client_1.expectString)(output.NextToken),
        StatusMessage: (0, smithy_client_1.expectString)(output.StatusMessage),
        Warnings: output.Warnings != null ? deserializeAws_json1_1Warnings(output.Warnings, context) : undefined,
    };
};
const deserializeAws_json1_1GetExpenseAnalysisResponse = (output, context) => {
    return {
        AnalyzeExpenseModelVersion: (0, smithy_client_1.expectString)(output.AnalyzeExpenseModelVersion),
        DocumentMetadata: output.DocumentMetadata != null
            ? deserializeAws_json1_1DocumentMetadata(output.DocumentMetadata, context)
            : undefined,
        ExpenseDocuments: output.ExpenseDocuments != null
            ? deserializeAws_json1_1ExpenseDocumentList(output.ExpenseDocuments, context)
            : undefined,
        JobStatus: (0, smithy_client_1.expectString)(output.JobStatus),
        NextToken: (0, smithy_client_1.expectString)(output.NextToken),
        StatusMessage: (0, smithy_client_1.expectString)(output.StatusMessage),
        Warnings: output.Warnings != null ? deserializeAws_json1_1Warnings(output.Warnings, context) : undefined,
    };
};
const deserializeAws_json1_1HumanLoopActivationOutput = (output, context) => {
    return {
        HumanLoopActivationConditionsEvaluationResults: output.HumanLoopActivationConditionsEvaluationResults != null
            ? new smithy_client_1.LazyJsonString(output.HumanLoopActivationConditionsEvaluationResults)
            : undefined,
        HumanLoopActivationReasons: output.HumanLoopActivationReasons != null
            ? deserializeAws_json1_1HumanLoopActivationReasons(output.HumanLoopActivationReasons, context)
            : undefined,
        HumanLoopArn: (0, smithy_client_1.expectString)(output.HumanLoopArn),
    };
};
const deserializeAws_json1_1HumanLoopActivationReasons = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        if (entry === null) {
            return null;
        }
        return (0, smithy_client_1.expectString)(entry);
    });
    return retVal;
};
const deserializeAws_json1_1HumanLoopQuotaExceededException = (output, context) => {
    return {
        Code: (0, smithy_client_1.expectString)(output.Code),
        Message: (0, smithy_client_1.expectString)(output.Message),
        QuotaCode: (0, smithy_client_1.expectString)(output.QuotaCode),
        ResourceType: (0, smithy_client_1.expectString)(output.ResourceType),
        ServiceCode: (0, smithy_client_1.expectString)(output.ServiceCode),
    };
};
const deserializeAws_json1_1IdempotentParameterMismatchException = (output, context) => {
    return {
        Code: (0, smithy_client_1.expectString)(output.Code),
        Message: (0, smithy_client_1.expectString)(output.Message),
    };
};
const deserializeAws_json1_1IdentityDocument = (output, context) => {
    return {
        DocumentIndex: (0, smithy_client_1.expectInt32)(output.DocumentIndex),
        IdentityDocumentFields: output.IdentityDocumentFields != null
            ? deserializeAws_json1_1IdentityDocumentFieldList(output.IdentityDocumentFields, context)
            : undefined,
    };
};
const deserializeAws_json1_1IdentityDocumentField = (output, context) => {
    return {
        Type: output.Type != null ? deserializeAws_json1_1AnalyzeIDDetections(output.Type, context) : undefined,
        ValueDetection: output.ValueDetection != null
            ? deserializeAws_json1_1AnalyzeIDDetections(output.ValueDetection, context)
            : undefined,
    };
};
const deserializeAws_json1_1IdentityDocumentFieldList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        if (entry === null) {
            return null;
        }
        return deserializeAws_json1_1IdentityDocumentField(entry, context);
    });
    return retVal;
};
const deserializeAws_json1_1IdentityDocumentList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        if (entry === null) {
            return null;
        }
        return deserializeAws_json1_1IdentityDocument(entry, context);
    });
    return retVal;
};
const deserializeAws_json1_1IdList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        if (entry === null) {
            return null;
        }
        return (0, smithy_client_1.expectString)(entry);
    });
    return retVal;
};
const deserializeAws_json1_1InternalServerError = (output, context) => {
    return {
        Code: (0, smithy_client_1.expectString)(output.Code),
        Message: (0, smithy_client_1.expectString)(output.Message),
    };
};
const deserializeAws_json1_1InvalidJobIdException = (output, context) => {
    return {
        Code: (0, smithy_client_1.expectString)(output.Code),
        Message: (0, smithy_client_1.expectString)(output.Message),
    };
};
const deserializeAws_json1_1InvalidKMSKeyException = (output, context) => {
    return {
        Code: (0, smithy_client_1.expectString)(output.Code),
        Message: (0, smithy_client_1.expectString)(output.Message),
    };
};
const deserializeAws_json1_1InvalidParameterException = (output, context) => {
    return {
        Code: (0, smithy_client_1.expectString)(output.Code),
        Message: (0, smithy_client_1.expectString)(output.Message),
    };
};
const deserializeAws_json1_1InvalidS3ObjectException = (output, context) => {
    return {
        Code: (0, smithy_client_1.expectString)(output.Code),
        Message: (0, smithy_client_1.expectString)(output.Message),
    };
};
const deserializeAws_json1_1LimitExceededException = (output, context) => {
    return {
        Code: (0, smithy_client_1.expectString)(output.Code),
        Message: (0, smithy_client_1.expectString)(output.Message),
    };
};
const deserializeAws_json1_1LineItemFields = (output, context) => {
    return {
        LineItemExpenseFields: output.LineItemExpenseFields != null
            ? deserializeAws_json1_1ExpenseFieldList(output.LineItemExpenseFields, context)
            : undefined,
    };
};
const deserializeAws_json1_1LineItemGroup = (output, context) => {
    return {
        LineItemGroupIndex: (0, smithy_client_1.expectInt32)(output.LineItemGroupIndex),
        LineItems: output.LineItems != null ? deserializeAws_json1_1LineItemList(output.LineItems, context) : undefined,
    };
};
const deserializeAws_json1_1LineItemGroupList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        if (entry === null) {
            return null;
        }
        return deserializeAws_json1_1LineItemGroup(entry, context);
    });
    return retVal;
};
const deserializeAws_json1_1LineItemList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        if (entry === null) {
            return null;
        }
        return deserializeAws_json1_1LineItemFields(entry, context);
    });
    return retVal;
};
const deserializeAws_json1_1NormalizedValue = (output, context) => {
    return {
        Value: (0, smithy_client_1.expectString)(output.Value),
        ValueType: (0, smithy_client_1.expectString)(output.ValueType),
    };
};
const deserializeAws_json1_1Pages = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        if (entry === null) {
            return null;
        }
        return (0, smithy_client_1.expectInt32)(entry);
    });
    return retVal;
};
const deserializeAws_json1_1Point = (output, context) => {
    return {
        X: (0, smithy_client_1.limitedParseFloat32)(output.X),
        Y: (0, smithy_client_1.limitedParseFloat32)(output.Y),
    };
};
const deserializeAws_json1_1Polygon = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        if (entry === null) {
            return null;
        }
        return deserializeAws_json1_1Point(entry, context);
    });
    return retVal;
};
const deserializeAws_json1_1ProvisionedThroughputExceededException = (output, context) => {
    return {
        Code: (0, smithy_client_1.expectString)(output.Code),
        Message: (0, smithy_client_1.expectString)(output.Message),
    };
};
const deserializeAws_json1_1Query = (output, context) => {
    return {
        Alias: (0, smithy_client_1.expectString)(output.Alias),
        Pages: output.Pages != null ? deserializeAws_json1_1QueryPages(output.Pages, context) : undefined,
        Text: (0, smithy_client_1.expectString)(output.Text),
    };
};
const deserializeAws_json1_1QueryPages = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        if (entry === null) {
            return null;
        }
        return (0, smithy_client_1.expectString)(entry);
    });
    return retVal;
};
const deserializeAws_json1_1Relationship = (output, context) => {
    return {
        Ids: output.Ids != null ? deserializeAws_json1_1IdList(output.Ids, context) : undefined,
        Type: (0, smithy_client_1.expectString)(output.Type),
    };
};
const deserializeAws_json1_1RelationshipList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        if (entry === null) {
            return null;
        }
        return deserializeAws_json1_1Relationship(entry, context);
    });
    return retVal;
};
const deserializeAws_json1_1StartDocumentAnalysisResponse = (output, context) => {
    return {
        JobId: (0, smithy_client_1.expectString)(output.JobId),
    };
};
const deserializeAws_json1_1StartDocumentTextDetectionResponse = (output, context) => {
    return {
        JobId: (0, smithy_client_1.expectString)(output.JobId),
    };
};
const deserializeAws_json1_1StartExpenseAnalysisResponse = (output, context) => {
    return {
        JobId: (0, smithy_client_1.expectString)(output.JobId),
    };
};
const deserializeAws_json1_1ThrottlingException = (output, context) => {
    return {
        Code: (0, smithy_client_1.expectString)(output.Code),
        Message: (0, smithy_client_1.expectString)(output.Message),
    };
};
const deserializeAws_json1_1UnsupportedDocumentException = (output, context) => {
    return {
        Code: (0, smithy_client_1.expectString)(output.Code),
        Message: (0, smithy_client_1.expectString)(output.Message),
    };
};
const deserializeAws_json1_1Warning = (output, context) => {
    return {
        ErrorCode: (0, smithy_client_1.expectString)(output.ErrorCode),
        Pages: output.Pages != null ? deserializeAws_json1_1Pages(output.Pages, context) : undefined,
    };
};
const deserializeAws_json1_1Warnings = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        if (entry === null) {
            return null;
        }
        return deserializeAws_json1_1Warning(entry, context);
    });
    return retVal;
};
const deserializeMetadata = (output) => {
    var _a;
    return ({
        httpStatusCode: output.statusCode,
        requestId: (_a = output.headers["x-amzn-requestid"]) !== null && _a !== void 0 ? _a : output.headers["x-amzn-request-id"],
        extendedRequestId: output.headers["x-amz-id-2"],
        cfId: output.headers["x-amz-cf-id"],
    });
};
const collectBody = (streamBody = new Uint8Array(), context) => {
    if (streamBody instanceof Uint8Array) {
        return Promise.resolve(streamBody);
    }
    return context.streamCollector(streamBody) || Promise.resolve(new Uint8Array());
};
const collectBodyString = (streamBody, context) => collectBody(streamBody, context).then((body) => context.utf8Encoder(body));
const buildHttpRpcRequest = async (context, headers, path, resolvedHostname, body) => {
    const { hostname, protocol = "https", port, path: basePath } = await context.endpoint();
    const contents = {
        protocol,
        hostname,
        port,
        method: "POST",
        path: basePath.endsWith("/") ? basePath.slice(0, -1) + path : basePath + path,
        headers,
    };
    if (resolvedHostname !== undefined) {
        contents.hostname = resolvedHostname;
    }
    if (body !== undefined) {
        contents.body = body;
    }
    return new protocol_http_1.HttpRequest(contents);
};
const parseBody = (streamBody, context) => collectBodyString(streamBody, context).then((encoded) => {
    if (encoded.length) {
        return JSON.parse(encoded);
    }
    return {};
});
const loadRestJsonErrorCode = (output, data) => {
    const findKey = (object, key) => Object.keys(object).find((k) => k.toLowerCase() === key.toLowerCase());
    const sanitizeErrorCode = (rawValue) => {
        let cleanValue = rawValue;
        if (typeof cleanValue === "number") {
            cleanValue = cleanValue.toString();
        }
        if (cleanValue.indexOf(":") >= 0) {
            cleanValue = cleanValue.split(":")[0];
        }
        if (cleanValue.indexOf("#") >= 0) {
            cleanValue = cleanValue.split("#")[1];
        }
        return cleanValue;
    };
    const headerKey = findKey(output.headers, "x-amzn-errortype");
    if (headerKey !== undefined) {
        return sanitizeErrorCode(output.headers[headerKey]);
    }
    if (data.code !== undefined) {
        return sanitizeErrorCode(data.code);
    }
    if (data["__type"] !== undefined) {
        return sanitizeErrorCode(data["__type"]);
    }
};
