// @aws-sdk/client-textract/dist-es/commands/AnalyzeIDCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AnalyzeIDRequestFilterSensitiveLog, AnalyzeIDResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1AnalyzeIDCommand, serializeAws_json1_1AnalyzeIDCommand } from "../protocols/Aws_json1_1";
var AnalyzeIDCommand = (function (_super) {
    __extends(AnalyzeIDCommand, _super);
    function AnalyzeIDCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AnalyzeIDCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TextractClient";
        var commandName = "AnalyzeIDCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AnalyzeIDRequestFilterSensitiveLog,
            outputFilterSensitiveLog: AnalyzeIDResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AnalyzeIDCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1AnalyzeIDCommand(input, context);
    };
    AnalyzeIDCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1AnalyzeIDCommand(output, context);
    };
    return AnalyzeIDCommand;
}($Command));
export { AnalyzeIDCommand };
