// @aws-sdk/client-textract/dist-es/commands/GetDocumentAnalysisCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetDocumentAnalysisRequestFilterSensitiveLog, GetDocumentAnalysisResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1GetDocumentAnalysisCommand, serializeAws_json1_1GetDocumentAnalysisCommand, } from "../protocols/Aws_json1_1";
var GetDocumentAnalysisCommand = (function (_super) {
    __extends(GetDocumentAnalysisCommand, _super);
    function GetDocumentAnalysisCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetDocumentAnalysisCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TextractClient";
        var commandName = "GetDocumentAnalysisCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetDocumentAnalysisRequestFilterSensitiveLog,
            outputFilterSensitiveLog: GetDocumentAnalysisResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetDocumentAnalysisCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetDocumentAnalysisCommand(input, context);
    };
    GetDocumentAnalysisCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetDocumentAnalysisCommand(output, context);
    };
    return GetDocumentAnalysisCommand;
}($Command));
export { GetDocumentAnalysisCommand };
